### PR TITLE
Migrate to ALLOW_***_BEGIN/END instead of #pragma clang diagnostic

### PR DESCRIPTION
**Pull request by litherum**
#### f204f5437c4e4edbe6551c2ad6dc58be653ba834
<pre>
Migrate to ALLOW_***_BEGIN/END instead of #pragma clang diagnostic
<a href="https://bugs.webkit.org/show_bug.cgi?id=255416">https://bugs.webkit.org/show_bug.cgi?id=255416</a>
rdar://108011779

Reviewed by Mike Wyrzykowski.

They&apos;re better because they&apos;re compiler-agnostic.

* Source/JavaScriptCore/disassembler/zydis/Zydis/ZycoreFormat.c:
* Source/WTF/wtf/darwin/OSLogPrintStream.mm:
(WTF::OSLogPrintStream::vprintf):
* Source/WebCore/platform/graphics/avfoundation/objc/CDMInstanceFairPlayStreamingAVFObjC.mm:
(WebCore::CDMInstanceFairPlayStreamingAVFObjC::contentKeySession):
(WebCore::CDMInstanceSessionFairPlayStreamingAVFObjC::ensureSessionOrGroup):
* Source/WebCore/platform/graphics/cg/NativeImageCG.cpp:
(WebCore::NativeImage::draw):
* Source/WebCore/platform/graphics/mac/controls/ControlMac.mm:
(WebCore::drawViewlessCell):
(WebCore::drawViewlessCellFocusRing):
* Source/WebCore/platform/mediastream/mac/ScreenCaptureKitCaptureSource.mm:
* Source/WebGPU/WebGPU/BindGroupLayout.mm:
(WebGPU::createArgumentDescriptor):
* Source/WebKit/Shared/ApplePay/cocoa/WebPaymentCoordinatorProxyCocoa.mm:
(WebKit::toPKShippingContactEditingMode):
* Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionCocoa.mm:
(WebKit::WebExtension::createError):
(WebKit::WebExtension::webProcessDisplayName):
* Source/WebKit/UIProcess/ios/WKContentView.h:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/AdditionalReadAccessAllowedURLs.mm:
(TEST):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/ContextMenus.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebViewConfiguration.mm:
(TEST):
* Tools/TestWebKitAPI/Tests/mac/WebViewIconLoading.mm:
(imageFromData):

Canonical link: <a href="https://commits.webkit.org/262993@main">https://commits.webkit.org/262993@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/f5e62a010b1f489ebd9d6bedb321d2cccf02a61c

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/3212 "10 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/3277 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/3388 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/4629 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/3563 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/3183 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/3341 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/3305 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/2795 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/3247 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/3560 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/2892 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/4472 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/1061 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/2863 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/2782 "layout-tests (failure)") | 
| [  ~~🛠 🧪 jsc~~](https://ews-build.webkit.org/#/builders/20/builds/2647 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/2841 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/2915 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/4190 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/3033 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/3292 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/2628 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/3284 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/2867 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/2865 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/798 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/793 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/2864 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/3369 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/3130 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/921 "Passed tests") |

### CODE REVIEW

**Review by djg — APPROVED** [review]
Unofficial :shipit: - Requires approval from reviewer.

File: Source/JavaScriptCore/disassembler/zydis/Zydis/ZycoreFormat.c
Comment (mwyrzykowski) [inline Source/JavaScriptCore/disassembler/zydis/Zydis/ZycoreFormat.c]: If this comes from Zycore-C should we be changing it?

Comment (aproskuryakov) [reply Source/JavaScriptCore/disassembler/zydis/Zydis/ZycoreFormat.c]: Also, it becomes inconsistent with the others.lines above.

Comment (litherum) [reply Source/JavaScriptCore/disassembler/zydis/Zydis/ZycoreFormat.c]: I'll revert this file.

**Review by mwyrzykowski — APPROVED** [review]

**Review by grorg — CHANGES_REQUESTED** [review]
It seems we don't indent these macros at the time of use.

File: Source/WTF/wtf/Assertions.cpp
Comment (mwyrzykowski) [inline Source/WTF/wtf/Assertions.cpp:196]: I guess this is opposite of what @grorg commented but every instance of `ALLOW_NONLITERAL_FORMAT_END` in WebKit currently is indented

I don't really care either way however

Comment (litherum) [reply Source/WTF/wtf/Assertions.cpp:196]: Before this patch, "\nALLOW_" has 420 hits in WebCore, whereas "    ALLOW_" has 258 hits. so I think Dean is actually right - we seem to be preferring the unindented form

Comment (litherum) [reply Source/WTF/wtf/Assertions.cpp:196]: Across the whole WebKit workspace, though, "\nALLOW_" has 833 hits, whereas " ALLOW_" has 1125.

So I guess it's not super clear what I should do. Maybe I should just leave the indentation alone?

Comment (litherum) [reply Source/WTF/wtf/Assertions.cpp:196]: After discussing offline, we're comfortable merging this patch as-is.

**Pull request merged**
